### PR TITLE
Add option to output check results in a tabular format 

### PR DESCRIPTION
**Pull request by mgarabed**
### What does this PR do?

This adds a new `--table` option to the `agent check` subcommand, allowing for the Series data to be formatted in a more condensed table.  It also sorts the metrics by name, and then by tags to help organize the output.

Example:

```
root@docker-desktop:/# agent check scylla --table
=== Series ===
  NAME                                                 TIMESTAMP   VALUE  TAGS                                                                                          
  scylla.storage.proxy.coordinator_read_latency.count  1606165366  110    [server:localhost:9180 shard:0 type:histogram upper_bound:1024.0]                             
  scylla.storage.proxy.coordinator_read_latency.count  1606165366  0      [server:localhost:9180 shard:0 type:histogram upper_bound:16.0]                               
  scylla.storage.proxy.coordinator_read_latency.count  1606165366  134    [server:localhost:9180 shard:0 type:histogram upper_bound:262144.0]                           
  scylla.storage.proxy.coordinator_read_latency.count  1606165366  134    [server:localhost:9180 shard:0 type:histogram upper_bound:32768.0]                            
  scylla.storage.proxy.coordinator_read_latency.count  1606165366  7      [server:localhost:9180 shard:1 type:histogram upper_bound:1048576.0]                          
  scylla.storage.proxy.coordinator_read_latency.count  1606165366  7      [server:localhost:9180 shard:1 type:histogram upper_bound:131072.0]                           
  scylla.storage.proxy.coordinator_read_latency.count  1606165366  7      [server:localhost:9180 shard:1 type:histogram upper_bound:262144.0]                           
  scylla.storage.proxy.coordinator_read_latency.count  1606165366  7      [server:localhost:9180 shard:1 type:histogram upper_bound:4096.0]                             
  scylla.storage.proxy.coordinator_read_latency.count  1606165366  7      [server:localhost:9180 shard:1 type:histogram upper_bound:65536.0]                            
```

### Motivation

Large metric output in JSON format becomes quite difficult to parse and grep easily without post processing scripts.  Formatting in a table can make the data easier to absorb quickly.

### Additional Notes

1. The sorting by tags is best effort as can be seen in the example above.  It will group appropriately by `shard` but the `upper_bound` tags are interpreted as strings and thus out of order.

1. This adds a new [external dependency](https://github.com/olekukonko/tablewriter), which seems relatively popular.  Of course, this could be also be coded manually to avoid the dependency if needed. 

1. This PR only addresses Series output, the other check output values such as Service Checks remain unchanged.

### CODE REVIEW

File: releasenotes/notes/tabular-check-format-4177ef9c7f8d0506.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Add `--table` option to `agent check` command to output
```
Comment (ofek) [inline releasenotes/notes/tabular-check-format-4177ef9c7f8d0506.yaml:11]:
```suggestion
    Add ``--table`` option to ``agent check`` command to output
```

**Review by ofek — CHANGES_REQUESTED** [review]
Awesome! WDYTA putting the timestamp at the end since we wouldn't need it usually?

File: pkg/metrics/series.go [hunk 2]
```diff
@@ -93,6 +94,40 @@ func (series Series) Marshal() ([]byte, error) {
...
 
+// MarshalStrings converts the timeseries to a sorted slice of string slices
+func (series Series) MarshalStrings() ([]string, [][]string) {
+	var headers = []string{"Metric", "Type", "Timestamp", "Value", "Tags"}
+	var payload = make([][]string, len(series))
+
+	for _, serie := range series {
+		payload = append(payload, []string{
+			serie.Name,
+			fmt.Sprintf(serie.MType.String()),
```
Comment (hush-hush) [inline pkg/metrics/series.go:105]: Do we need Sprintf ?

File: pkg/metrics/series.go [hunk 3]
```diff
@@ -93,6 +94,40 @@ func (series Series) Marshal() ([]byte, error) {
...
+			fmt.Sprintf("%.0f", serie.Points[0].Ts),
+			fmt.Sprint(serie.Points[0].Value),
```
Comment (hush-hush) [inline pkg/metrics/series.go:107]: Same here: do we need Sprintf

File: pkg/metrics/series.go [hunk 4]
```diff
@@ -93,6 +94,40 @@ func (series Series) Marshal() ([]byte, error) {
...
+			fmt.Sprint(serie.Tags),
```
Comment (hush-hush) [inline pkg/metrics/series.go:108]: I think `strings.Join` would produce a more readable result. WDYT ?

**Review by hush-hush — CHANGES_REQUESTED** [review]

File: pkg/metrics/series.go (hunk 3 above)
Comment (mgarabed) [reply pkg/metrics/series.go:107]: What would be an alternative to convert a float64 value into a string?  Leaving it as `Value` results in an error for the string slice. 

(New to Go so could be missing something obvious!)

Comment (ofek) [reply pkg/metrics/series.go:107]: May help: https://golang.org/pkg/strconv/

Comment (mgarabed) [reply pkg/metrics/series.go:107]: Looks like `strconv.FormatFloat(serie.Points[0].Value, 'f', -1, 64)` is about 40% faster.

**Comment by mgarabed** [comment]
> Awesome! WDYTA putting the timestamp at the end since we wouldn't need it usually?

@ofek The only issue with that would be the alignment of the columns.  Since tags are so variable they can extend the line considerably, and having timestamps at the end would muddy the output and make it harder to see things at a glance.

**Review by hush-hush — APPROVED** [review]

**Review by hush-hush — APPROVED** [review]

**Review by ofek — APPROVED** [review]

**Pull request merged**